services/slam/builtin/builtin_test.go

```diff
@@ -683,15 +683,14 @@ func TestCartographerDataProcess(t *testing.T) {
 		}()
 
 		cancelCtx, cancelFunc := context.WithCancel(context.Background())
-		slamSvc.StartDataProcess(cancelCtx, cams, camStreams)
+		c := make(chan int)
+		slamSvc.StartDataProcess(cancelCtx, cams, camStreams, c)
 
-		n := 5
-		// Note: timePadding is required to allow the sub processes to be fully completed during test
-		time.Sleep(time.Millisecond * time.Duration((n)*(validDataRateMS+timePadding)))
+		fileSaved := <-c
+		test.That(t, fileSaved, test.ShouldEqual, 1)
```

Comment:
[nit] No need to check the value, since this channel is just used for signaling.